עד מתי אוג 20

imoreno1 opened the pull request.

oorielmoreno committed:
עד מתי אוג 20

ItayReshef commented on pythonProject1/main.py:
```diff
@@ -18,19 +17,4 @@
         answer = solution
     is_primal = True
     solution += 1
```
Variables outside of functions or classes should be consts. (All Caps)

imoreno1 replied:
fuck off

ItayReshef commented on pythonProject1/main.py:
```diff
@@ -18,19 +17,4 @@
         answer = solution
     is_primal = True
     solution += 1
```
variables should have type decleration

ItayReshef commented on pythonProject1/main.py:
```diff
@@ -1,6 +1,5 @@
+def is_prime(number: int)-> bool:
 
 subject = 600851475143
 solution = 9
```
use better naming - something that recalls and relates to the question and allows you to make more sense of oyur variables

oorielmoreno committed:
למה זה לא עובד סעמק

ItayReshef commented on pythonProject1/main.py:
```diff
@@ -1,36 +1,49 @@
+TARGET_NUMBER = 13195
 
+def largest_prime_factor(subject: int)-> int:
+    minimal_possible_answer = round(subject / 2)
```
לדעתי צדקת ואפשר להפוך את זה פה לשורש של המספר

imoreno1 replied:
בסוף ככה עשיתי וזה עבד אחלה, צריך רק לעגל

ItayReshef commented on pythonProject1/main.py:
```diff
@@ -1,36 +1,49 @@
     while primal_checker > 1:
-        if solution % primal_checker == 0:
+        if number % primal_checker == 0:
             is_primal = False
```
instead of using a boolean variable and breaking you could instead return False here which will break the loop and the function. And return True by default at the end of this function.

imoreno1 replied:
לגיטימי

ItayReshef commented on pythonProject1/main.py:
```diff
@@ -1,36 +1,49 @@
-while solution < subject:
-    primal_checker = solution - 1
+        for y in reversed(range(minimal_possible_answer)):
+            if minimal_possible_answer * multiplier == subject and is_prime(minimal_possible_answer):
```
You should use for loops in a different manner. 
use the iterator in the function(y) - do no use what you iterating on (minimal_possible_answer)

ItayReshef commented on pythonProject1/main.py:
```diff
@@ -1,36 +1,49 @@
-primal_checker = 0
-while solution < subject:
-    primal_checker = solution - 1
+        for y in reversed(range(minimal_possible_answer)):
```
y - bad naming

imoreno1 replied:
דוקטור דוקטור would like to disagree

imoreno1 replied:
אבל הוא לא דוגמא לכלום

ItayReshef commented on pythonProject1/main.py:
```diff
@@ -1,36 +1,49 @@
+        for y in reversed(range(minimal_possible_answer)):
+            if minimal_possible_answer * multiplier == subject and is_prime(minimal_possible_answer):
+                return minimal_possible_answer
+            multiplier -= 1
+        minimal_possible_answer -= 1
```
If i'm not mistaken both of them remain exactly the same throguht this loop.
Why are they both neccessary?

ItayReshef commented on pythonProject1/fibonacci.py:
```diff
@@ -0,0 +1,10 @@
+maxval = 1
+fibonacci = [1, 2]
+result = 0
+while fibonacci[len(fibonacci) - 1] < 4000000:
+    fibonacci.append(fibonacci[maxval-1]+fibonacci[maxval])
+    maxval = maxval+1
+for y in fibonacci:
+    if y%2 == 0:
+        result += y
+print(result)
```
implement everything here too if you want anothther review on it

ItayReshef commented on pythonProject1/main.py:
```diff
@@ -1,36 +1,49 @@
+
+def is_prime(number: int)-> bool:
+    is_primal = True
+    primal_checker = number - 1
```
You could use a more efficient starting number for the primal checker.
better naming offering - "denominator"

ItayReshef reviewed (commented):
תנסה לספר יותר לעצמך את הסיפור של מה אמור לקרות עם המספרים ולמה זה יעבוד אחרי שתבצע את הפעולות האלה כמו שהן - אם הסיפור לא עובד לך כנראה זה לא יעבוד גם בקוד.

oorielmoreno committed:
עובד :)

oorielmoreno committed:
File "F:\GIt\moreno_study\pythonProject1\palindrome.py", line 11, in is_palindrome
    if int_list[list_index] != int_list[len(int_list -list_index)]:
TypeError: unsupported operand type(s) for -: 'list' and 'int'

 ^^^^^^^
 איך זה הגיוני, אני קורא לאינדקסים ברשימה של אינטים מה הוא רוצה ממני

ItayReshef commented:
You are trying to diminish an index of an list from a list. What do you think is going to happen?

oorielmoreno committed:
le solved

oorielmoreno committed:
le returns 0 idk why karich help

imoreno1 commented:
הבאתי בclone וממשיך לעבוד בבסיס
